parsl/monitoring/visualization/plots/default/workflow_plots.py

```diff
@@ -197,25 +199,26 @@ def workflow_dag_plot(df_tasks, group_by_apps=True):
             name=k,          # legend app_name here
             marker=dict(
                 showscale=False,
-                # color='rgb(200,0,0)',
+                color=color,
                 size=11,
                 line=dict(width=1, color='rgb(0,0,0)')))
-        node_traces.append(node_trace)
+        node_traces[index]=node_trace
 
     for node in node_positions:
         x, y = node_positions[node]
         if group_by_apps:
             name = dic['task_func_name'][node]
-            index = groups_list[name]
+            index, color = groups_list[name]
```

Comment:
We can probably move this line outside of the `if ... else` now. And also, after removing line 218, `color` is no longer used later I think.